src/flanders/example.cljc

```diff
@@ -32,63 +31,66 @@
   ;; Branches
 
   EitherType
-  (->example [{:keys [choices]} f]
-    (f (first choices)))
+  (->example [{:keys [choices] :as ddl} f]
+    (let [[_ example :as example?] (find ddl :default)]
+      (if example?
+        example
+        (f (first choices)))))
 
   MapEntry
-  (->example [{:keys [key type]} f]
-    [(f (assoc key
-               :key? true))
-     (f type)])
+  (->example [{:keys [key type] :as ddl} f]
+    (let [[_ example :as example?] (find ddl :default)]
+      [(f (assoc key :key? true))
+       (f (cond-> type
+            example? (assoc :default example)))]))
 
   MapType
-  (->example [{:keys [entries]} f]
-    (reduce (fn [m [k v]]
-              (assoc m k v))
-            {}
-            (map f entries)))
+  (->example [{:keys [entries default]} f]
+    (or default
+        (reduce (fn [m [k v]]
+                  (assoc m k v))
+                {}
+                (map f entries))))
 
   SequenceOfType
-  (->example [{:keys [type]} f]
-    [(f type)])
+  (->example [{:keys [type default]} f]
+    (or default [(f type)]))
 
   SetOfType
-  (->example [{:keys [type]} f]
-    #{(f type)})
+  (->example [{:keys [type default]} f]
+    (or default #{(f type)}))
 
   ;; Leaves
 
   AnythingType
-  (->example [_ _]
-    {:anything "anything"})
+  (->example [{:keys [default]} _]
+    (or default "anything"))
```

Comment:
I am not sure if `AnythingType` can be `nil`. If that's the case, I think you should use `find` here instead of just `or`.


```clojure
(-> example [m _]
  (if-let [[_ v] (find m :default)]
     v
     "anything"))
```